src/Background/estimators.jl

```diff
@@ -133,3 +133,60 @@ end
 
 MMM() = MMM(3, 2)
 estimate_background(alg::MMM, data; dims = :) = alg.median_factor * median(data, dims = dims) - alg.mean_factor * mean(data, dims = dims)
+
+"""
+    BiweightLocation(c = 6.0, M = nothing)
+
+Estimate the background using the robust biweight location statistic.
+
+``\\xi_{biloc}=M + \\frac{\\sum_{|u_i|<1}{(x_i - M)(1 - u_i^2)^2}}{\\sum_{|u_i|<1}{(1-u_i^2)^2}}``
+with
+``u_i = \\frac{(x_i - M)}{c\\cdot\\text{MAD}(x)}``
+here ``\\text{MAD}(x)`` is median absolute deviation of `x`.
+
+# Example
+```jldoctest
+julia> x = ones(3,5);
+
+julia> estimate_background(BiweightLocation, x)
+1.0
+
+julia> estimate_background(BiweightLocation(5.5), x; dims = 1)
+1×5 Array{Float64,2}:
+ 1.0  1.0  1.0  1.0  1.0
+ ```
+"""
+struct BiweightLocation{T <: Number} <: BackgroundEstimator
+    c::T
+    M::Union{Nothing, T}
+end
+
+BiweightLocation(c; M = nothing) = BiweightLocation(c, M)
+BiweightLocation() = BiweightLocation(6.0)
+
+function biweight_location(data::AbstractArray, c, M=median(data))
+
+    M = M === nothing ? median(data) : M
+
+    MAD = mad(data, normalize = false)
+
+    MAD ≈ 0 && return M
+
+    u = @. (data - M) / (c * MAD)
+
+    num = 0
+    den = 0
```

Comment:
Even without running the code I'm pretty sure that these are type unstable, please see https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-changing-the-type-of-a-variable-1.  They can probably initialise with something like `zero(eltype(u))`.

I strongly recommend using [`@code_warntype`](https://docs.julialang.org/en/v1/manual/performance-tips/#man-code-warntype-1) to check whether the functions are type stable or not.  Type instabilities in Julia can brutally kill performance.

As a general advice, get familiar with all [performance tips](https://docs.julialang.org/en/v1/manual/performance-tips/)